File: cc_inheritance.java
import java.util.*;
class Base{
  void fun()
  {
    System.out.println("This is from Base Class\n");
  }
}
class Derived1 extends Base{
  void fun()
  {
    System.out.println("From D1 Class");
  }
}
class Derived2 extends Base{
  void fun()
  {
  System.out.println("From D2 class\n");
  }
  void fun(int a)
  {
  System.out.println("From D2 with value "+a);
  }
}
class MainClass{
public static void main(String arg[])
{
  Derived1 d1=new Derived1();
  d1.fun(1);
}
}

File: cc_inheritance.cpp
#include<iostream>
#include<cstdio>
using namespace std;
class Base{
public :
  virtual void fun()
  {
    cout<<"From Base Class\n";
  }
} ; 
class Derived1 : public Base
{
};
class Derived2 : public Base{

public :
  void fun()
  { 
    cout<<"From D2 Class\n";
  }
  void fun(int a)
  {
    cout<<"This is from D2 with value "<<a<<endl;
  }
};
int main()
{
  Base *B;
  Derived2 D1;
  B=&D1;
  B->fun();
  return 0;
}